### DISCUSSION TITLE
For Fuseki 4.6.1, how to load and export large RDF files without splitting the files?

### DISCUSSION THREAD

**Discussion by chenkuncloud** [apache/jena#1554] [Q&A] [answered]
Hi,

I have two problems:

1. How to load large RDF files(2G - 10G) into Fuseki 4.6.1 without splitting the files?
2. How to export all RDF triples from Fuseki into a file(more than 5 million RDF triples)?

Both HTTP and command line are acceptable.

I learned that Virtuoso can be implemented using [ISQL](https://vos.openlinksw.com/owiki/wiki/VOS/VirtBulkRDFLoader). Does Fuseki has a similar approach?

**Comment by afs** [comment, marked as answer]
Q1: Upload
Using a TDB2 database:

Method 1: POST the file (e.g. with curl) including content type header.
Method 2: uses the TDB2 loader and build a database offline, then put it in place.

5-10 million triples isn't big - method 1 should be fine and has less manual steps.

Q2: Download
Method 1: Fuseki has a backup operation - it writes a file to the servers local disk.
Method 2: Do a HTTP GET: query to get everything (make sure the client software does not buffer responses). 
For example: `curl http://host:3030/dataset`. The default is NQuads. Content-negotation (`Accept` header) can be used.

**Reply by chenkuncloud** [reply]
Thanks.

Best Regards
